helix-core/src/main/java/org/apache/helix/controller/stages/MessageOutput.java

```diff
@@ -64,6 +69,14 @@ public void addMessages(String resourceName, Partition partition,
     return Collections.emptyList();
   }
 
+  public Map<Partition, List<Message>> getResourceMessages(String resourceName) {
```

Comment:
This method could be costly since it needs to read and create an unmodifiable view for all the messages for a resource. I think getMessages with resource name and partition shall be working fine. Can we use that instead?

Reply:
We talked about this before. 
The usage is `selectedMessage.getResourceMessages(resourceName)`. Without adding this new API, how can you construct `Map<Partition, List<Message>>`? 

One have to do something like this:

```
Map<Partition, List<Message>> partitonMsgMap = new HashMap();
for (Partition partition : resource.getPartitions()) {
   List<Message> msgList = selectedMessage.getMessage( resource, parttiton);
   partitionMsgMap.put(partition, messages);
}

```
This would loop against all partition for resource. From CPU cycle, it won't save anything. 
Also the code seems to be bloated to me.

This is not essential. If you think this way is better, we can also do it this way. Or there is another way in your mind, please draw the code out. If it works, we can also use it that way.

Reply:
We don't need to get the whole map. I think you can just pass the output to the user method and then call the existing method to fetch the list for each resource/partition.

Reply:
removed getResourceMessages.